src/com/itacademy/eshop/shop/Eshop.java

```diff
@@ -25,8 +26,15 @@ public void printProducts() {
         }
     }
 
-    public void addProduct(Product shirt) {
-        products.add(shirt);
+    public void addProduct(Product shirt) throws DuplicateProductException {
```

Comment:
i dont think this method works as intended.
in this case it iterates through all products in the list, and for EACH element in the list, if it does not match, it gets added, and the exception is thrown regardless. this means that if there are 50 items that are not "shirts", if you try to add a shirt, you would add 50 shirts AND throw a DuplicateProductException

Reply:
also should use a more generic name than "shirt"